avocado_vt/test.py

```diff
@@ -395,17 +395,21 @@ def _runTest(self):
             test_modules[t_type] = imp.load_module(t_type, f, p, d)
             f.close()
 
-        # TODO: the environment file is deprecated code, and should be removed
-        # in future versions. Right now, it's being created on an Avocado temp
-        # dir that is only persisted during the runtime of one job, which is
-        # different from the original idea of the environment file (which was
-        # persist information accross virt-test/avocado-vt job runs)
```

Comment:
And as someone that agrees, do you have a suggestion for an alternative to achieve persistence of the vms?

The TODO removed here is entirely deprecated if we are to restore optional vm persistence and the TODO states *exactly* that it has to be removed since it is a remnant of `virt-test`'s persistence option.

Reply:
And as I mentioned in other places in your review like https://github.com/avocado-framework/avocado-vt/pull/2105#issuecomment-573187772, this persistence is needed for our own plugin for plenty of reasons. I don't see why we can't make this optional for those of us that need the extra flexibility - whenever the `avocado-vt` maintainers considered only permanence within jobs they might not have had in mind all the possible use cases and reasonable extensions. It is a normal part of the development process.

Reply:
Yep, removing the TODO is fine. We wanted to remove this because keeping VMs running after test is tricky, keeping it after job is even trickier and there always were some issues like leaked fds, duplicate VMs or different VMs pretending to be the same.

With this said we don't have an alternative and as far as I know nobody works on changing this. And if you are brave and know what you want to do (keep VMs after job finishes) the deprecation note needs to be removed.

Reply:
So far we have been using this feature for 7 years, all the way from `virt-test` to patched `avocado-vt` :) We have seen all of this but in the last years things have been fairly clean if we exclude an occasional glitch in LVM that we use for taking vm snapshots and reusing them for tests. This being said, I will resolve this then and summarize everything left from these reviews below.